configuration/pih/liquibase/sql/void_duplicate_registrations.sql

```diff
@@ -0,0 +1,158 @@
+name                              varchar(50),   
+family_name                       varchar(50),   
+dob                               date,          
+gender                            varchar(2),    
+PRIMARY KEY (group_id)
+);
+
+-- load table with registration info of all the duplicates, using group_id to tie together the groups of dups 
+INSERT INTO temp_dup_info(name, family_name, dob, gender, mothers_first_name, telephone_number, address_level_1, address_level_2, address_level_3, address_level_4, address_level_5, reg_location, registration_date, registration_entry_date, user_entered)
+ select name, family_name, dob, gender, mothers_first_name, telephone_number, address_level_1, address_level_2, address_level_3, address_level_4, address_level_5, reg_location, registration_date, DATE(registration_entry_date), user_entered
+ from temp_patients
+ group by name, family_name, dob, gender, mothers_first_name, telephone_number, address_level_1, address_level_2, address_level_3, address_level_4, address_level_5, reg_location, registration_date, DATE(registration_entry_date), user_entered
+ having count(*) > 1
+ ;
+
+-- join together the dup information with the entire list of patients
+-- to create a list of patients with patient_ids grouped together
+DROP TEMPORARY TABLE IF EXISTS temp_potential_dups;
+CREATE TEMPORARY TABLE  temp_potential_dups
+select d.group_id, p.* from temp_patients p
+inner join temp_dup_info d
+on d.name <=> p.name -- -- note '<=>' will compare nulls properly
+and d.family_name <=> p.family_name
+and d.dob <=> p.dob
+and d.gender <=> p.gender
+and d.mothers_first_name <=> p.mothers_first_name
+and d.telephone_number <=> p.telephone_number
+and d.address_level_1 <=> p.address_level_1
+and d.address_level_2 <=> p.address_level_2
+and d.address_level_3 <=> p.address_level_3
+and d.address_level_4 <=> p.address_level_4
+and d.address_level_5 <=> p.address_level_5
+and d.reg_location <=> p.reg_location
+and d.registration_date <=> p.registration_date
+and date(d.registration_entry_date) <=> date(p.registration_entry_date)
+and d.user_entered <=> p.user_entered;
+
+DROP TEMPORARY TABLE IF EXISTS temp_potential_dups_patient_ids;
+CREATE TEMPORARY TABLE  temp_potential_dups_patient_ids
+select group_id, patient_id from temp_potential_dups;
+
+-- create final list of patients to void by selecting the patients perviously chosen who:
+-- do not have any encounters but where the other patients in that group have encounters (excluding registration)   
+DROP TEMPORARY TABLE IF EXISTS temp_dups_to_void;
+CREATE TEMPORARY TABLE  temp_dups_to_void
+select patient_id from temp_potential_dups d 
+where not exists
+ (select 1 from encounter e where  e.patient_id = d.patient_id and e.encounter_type <> @regEncounterType)
+and exists 
+ (select 1 from encounter e2 where e2.patient_id in
+ 	(select patient_id from temp_potential_dups_patient_ids d2 where d2.group_id = d.group_id)
+ 	and e2.encounter_type <> @regEncounterType
```

Comment:
I'm probably missing this, but do you need  to confirm that e2.patient is not equalt to e.patient?  Or I guess it's the fact that the two clauses can never be both true for the same patient?

Reply:
right... I didn't think to do it because it would need to be a different patient for the two clauses to be true.  Shouldn't need it.